### PR TITLE
HTFA 2-level cross validation example

### PR DESCRIPTION
**Pull request by xiazhu**

### CODE REVIEW

**Comment by mihaic** [comment]
You must also merge the latest commit from master which fixed a bug that was causing the tests to fail.

File: examples/factoranalysis/htfa_cv_example.py [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+#get fMRI data and scanner RAS coordinates
+data = []
+R = []
+mapping = {}
+n_local_subj = 0
+for idx in range(n_subj):
+    if idx % size == rank:
+        mapping[str(n_local_subj)] = idx
+        n_local_subj += 1
+        file_name = os.path.join(data_dir, 's' + str(idx) + '.mat')
```
Comment (mihaic) [inline examples/factoranalysis/htfa_cv_example.py]: I actually meant something much simpler: check if the file exist in the filesystem so the user can re-run the example without re-downloading the files every time.

File: examples/factoranalysis/htfa_cv_example.py [hunk 2]
```diff
@@ -58,8 +59,8 @@ def get_train_err(htfa, data, F):
     Parameters
     ----------
 
-    htfa : HTFA 
-        Factor Anaysis class in Brainiak.
+    htfa : HTFA
+        An instance of HTFA, factor anaysis class in Brainiak.
```
Comment (mihaic) [inline examples/factoranalysis/htfa_cv_example.py]: We should keep the spelling consistent: BrainIAK.

File: examples/factoranalysis/run_mpi_htfa_cv.sh [hunk 3]
```diff
@@ -1,3 +1,5 @@
 #!/bin/sh
 
-mpirun -np 6 python3 htfa_cv_example.py 
+#the number of process is configurable, 6 is just an example
+NUM_PROCESS=6
```
Comment (mihaic) [inline examples/factoranalysis/run_mpi_htfa_cv.sh]: How about this?
`NUM_PROCESS=${1:-6}`

File: examples/factoranalysis/htfa_cv_example.py [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+
+#get fMRI data and scanner RAS coordinates
+data = []
+R = []
+mapping = {}
+n_local_subj = 0
+for idx in range(n_subj):
+    if idx % htfa_size == htfa_rank:
+        file_name = os.path.join(data_dir, 's' + str(idx) + '.mat')
+        all_data = scipy.io.loadmat(file_name)
```
Comment (mihaic) [inline examples/factoranalysis/htfa_cv_example.py]: I tried the example and noticed some file not found errors:
```
$ ./run_mpi_htfa_cv.sh 
About to run command 'mpirun -np 6 python3 htfa_cv_example.py'
Traceback (most recent call last):
  File "htfa_cv_example.py", line 185, in <module>
    all_data = scipy.io.loadmat(file_name)
  File "/home/mihai/.local/lib/python3.5/site-packages/scipy/io/matlab/mio.py", line 135, in loadmat
    MR = mat_reader_factory(file_name, appendmat, **kwargs)
  File "/home/mihai/.local/lib/python3.5/site-packages/scipy/io/matlab/mio.py", line 58, in mat_reader_factory
    byte_stream = _open_file(file_name, appendmat)
  File "/home/mihai/.local/lib/python3.5/site-packages/scipy/io/matlab/mio.py", line 23, in _open_file
    return open(file_like, 'rb')
FileNotFoundError: [Errno 2] No such file or directory: '/home/mihai/Surse/brainiak/examples/factoranalysis/data/s0.mat'
```
I think we need a barrier before this loop, so the threads not involved in the download wait for the download to finish.

Comment (xiazhu) [inline examples/factoranalysis/htfa_cv_example.py]: done.

**Comment by mihaic** [comment]
@xiazhu, note that I merged the latest changes from master to your branch using the "Update branch" button on github.com. Before adding more changes to your local version, pull these changes. In the future, you should consider using feature-specific branches for PRs that can be deleted merging, instead of the master branch.

**Comment by xiazhu** [comment]
I have always been using branches for code development. For this particular one, i had it fully working on another machine, so did not bother to create another branch for it.  For future PRs developing from the scratch, i'll use branches.

**Pull request merged**